uart_example/lib/main.dart

```diff
@@ -0,0 +1,342 @@
+    PermissionStatus permission;
+    permission = await LocationPermissions().requestPermissions();
+    if (permission == PermissionStatus.granted) {
+      _foundBleUARTDevices = [];
+      _scanning = true;
+      refreshScreen();
+      _scanStream =
+          flutterReactiveBle.scanForDevices(withServices: [_UART_UUID]).listen((
+              device) {
+            if (_foundBleUARTDevices.every((element) =>
+            element.id != device.id)) {
+              _foundBleUARTDevices.add(device);
+              refreshScreen();
+            }
+          }, onError: (Object error) {
+            _logTexts =
+                _logTexts + "ERROR while scanning:" + error.toString() + " \n";
+            refreshScreen();
+          }
+          );
+    }
+    else {
+      await showNoPermissionDialog();
+    }
+  }
+
+  void onConnectDevice(index) {
+    _currentConnectionStream = flutterReactiveBle.connectToAdvertisingDevice(
+      id:_foundBleUARTDevices[index].id,
+      prescanDuration: Duration(seconds: 1),
+      withServices: [_UART_UUID, _UART_RX, _UART_TX],
+    );
+    _logTexts = "";
+    refreshScreen();
+    _connection = _currentConnectionStream.listen((event) {
+      String id = event.deviceId.toString();
+      switch(event.connectionState) {
+        case DeviceConnectionState.connecting:
+          {
+            _logTexts = _logTexts + "Connecting to " + id +"\n";
+            break;
+          }
+        case DeviceConnectionState.connected:
+          {
+            _connected = true;
+            _logTexts = _logTexts + "Connected to " + id + "\n";
+            _numberOfMessagesReceived = 0;
+            _receivedData = [];
+            _txCharacteristic = QualifiedCharacteristic(serviceId: _UART_UUID, characteristicId: _UART_TX, deviceId: event.deviceId);
+            _receivedDataStream = flutterReactiveBle.subscribeToCharacteristic(_txCharacteristic);
+            _receivedDataStream.listen((data) {
+               onNewReceivedData(data);
+            }, onError: (dynamic error) {
+              _logTexts = _logTexts + "Error:" + error.toString() + id + "\n";
+            });
+            _rxCharacteristic = QualifiedCharacteristic(serviceId: _UART_UUID, characteristicId: _UART_RX, deviceId: event.deviceId);
+            break;
+          }
+        case DeviceConnectionState.disconnecting:
+          {
+            _connected = false;
+            _logTexts = _logTexts + "Disconnecting from " + id + "\n";
+            break;
+          }
+        case DeviceConnectionState.disconnected:
+          {
+            _logTexts = _logTexts + "Disconnected from " + id + "\n";
+            break;
+          }
+      }
+      refreshScreen();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        // Here we take the value from the MyHomePage object that was created by
+        // the App.build method, and use it to set our appbar title.
```

Comment:
The next line explains it in four "words" of code, I'd say. Consider removing the comment.